karate-core/pom.xml

```diff
@@ -69,7 +70,12 @@
             <groupId>info.picocli</groupId>
             <artifactId>picocli</artifactId>
             <version>4.0.3</version>
-        </dependency>                     
+        </dependency>
```

Comment:
@michaelpro1  I'm really sorry that I took so much time to get to review this, I was heads down doing windows app automation - and I should have flagged this right at the start - apologies

a lot of work has gone into keeping the dependencies of `karate-core` minimal, e.g. no commons dependencies, home grown string utils etc. and we take care to maven-shade netty and antlr as it typically surfaces in the wild and can cause lib conflicts

is there any way we can avoid this dependency or you could try "shade" it (refer the karate-core pom.xml)

I've noticed that jersey libs for some reasons don't "shade well" because they do some funky DI / reflection and what not, so watch out for that

Reply:
Ah I see, I can do that. 
Another alternative is to fork the `UriTemplateParser` which is pretty much self-contained apart from Error Messages and some hexCode references however that would mean that would have to be manually maintained. Plus side of this is that the size of the JAR will not be big.
Which one do you prefer?

Reply:
I would prefer the fork and duplicating code. if you run into challenges, we can try include the JAR dependency but it has to be maven-shaded. else we can think of other options

Reply:
OK done, needed one extra dependency as JDK one did not work. Please review.